tests/lib/c_lib/src/main.c

```diff
@@ -319,6 +322,334 @@ void test_bsearch(void)
+
+	ret = memchr(str, ch2, strlen(str));
+	zassert_is_null(ret, "memchr 'testfunction' 'z'");
+}
+
+/**
+ *
+ * @brief Test memcpy function
+ *
+ */
+void test_memcpy(void)
+{
+	char arr[10];
+	char *str = "12345";
+	int len = 5;
+
+	memcpy(arr, str, len + 1);
+	zassert_true((memcmp(str, arr, len + 1)) == 0,
+		     "memcpy failed");
+}
+
+/**
+ *
+ * @brief test memmove function
+ *
+ */
+void test_memmove(void)
+{
+	char str1[7] = "aabbcc";
+	char str2[7] = "aaaabb";
+
+	memmove(str1 + 2, str1, 4);
+	zassert_true(memcmp(str1, str2, 7) == 0,
+		     "memmove failed");
+}
+
+/**
+ *
+ * @brief test strcat function
+ *
+ */
+void test_strcat(void)
+{
+	char str[20] = "hello";
+	char src[20] = "world";
+
+	strcat(str, src);
+	zassert_true(strcmp(str, "helloworld") == 0, "test strcat failed");
+}
+
+/**
+ *
+ * @brief test strcspn function
+ *
+ */
+void test_strcspn(void)
+{
+	char *str1 = "aabbcc";
+	char *str2 = "b";
+	char *str3 = "d";
+	int ret;
+
+	ret = strcspn(str1, str2);
+	zassert_true((ret == 2), "strcspn failed");
+	ret = strcspn(str1, str3);
+	zassert_true((ret == 6), "strcspn not found str");
+}
+
+/**
+ *
+ * @brief test strncat function
+ *
+ */
+void test_strncat(void)
+{
+	char str[20] = "aa";
+	char src[20] = "bbccdd";
+
+	strncat(str, src, 2);
+	zassert_true((strcmp(str, "aabb")) == 0, "strncat failed");
+}
+
+/**
+ *
+ * @brief test strrchr function
+ *
+ */
+void test_strrchr(void)
+{
+	char str[] = "abcccbf";
+	char *ptr;
+
+	zassert_is_null(strrchr(str, 'd'),
+		       "strrchr not found this word. failed");
+	ptr = strrchr(str, 'b');
+	zassert_true((strcmp(ptr, "bf") == 0), "strrchr failed");
+}
+
+/**
+ *
+ * @brief test strstr function
+ *
+ */
+void test_strstr(void)
+{
+	char *str1 = "aabbccdd";
+	char *str2 = "aa";
+	char *str3 = "ff";
+
+	zassert_is_null(strstr(str1, str3), "strstr aabbccdd with ff failed");
+	zassert_not_null(strstr(str1, str2), "strstr aabbccdd with aa failed");
+}
+
+/**
+ *
+ * @brief test strtol function
+ *
+ */
+void test_strtoxl(void)
+{
+	char buf1[20] = "10379aegi";
+	char buf2[20] = "10379aegi";
+	char buf3[20] = "010379aegi";
+	char buf4[20] = "0x10379aegi";
+	char *stop = NULL;
+	long ret;
+	unsigned long retul;
+
+	ret = strtol(buf3, &stop, 8);
+	zassert_true((ret == 543), "strtol base = 8 failed");
+	ret = strtol(buf1, &stop, 10);
+	zassert_true((ret == 10379), "strtol base = 10 failed");
+	retul = strtoul(buf2, &stop, 10);
+	zassert_true((ret == 10379), "strtoul base = 10 failed");
+	ret = strtol(buf4, &stop, 16);
+	zassert_true((ret == 17004974), "strtol base = 16 failed");
+}
+
+/**
+ *
+ * @brief test tolower function
+ *
+ */
+void test_tolower(void)
+{
+	int str1 = 'A';
+	int str2 = 'Z';
+	char lower;
+	char *sptr;
+
+	lower = tolower(str1);
+	sptr = &lower;
+	zassert_true((strncmp(sptr, "a", 1) == 0), "tolower 'a' failed");
+
+	lower = tolower(str2);
+	sptr = &lower;
+	zassert_true((strncmp(sptr, "z", 1) == 0), "tolower 'z' failed");
```

Comment:
I don't understand why you are doing this.

Reply:
Oh, it is my mistake. I will making change for this with another assert.